dlt/pipeline/helpers.py

```diff
@@ -152,58 +133,6 @@ def is_empty(self) -> bool:
             and len(self.info["resource_states"]) == 0
         )
 
-    def _drop_destination_tables(self) -> None:
-        table_names = [tbl["name"] for tbl in self.tables_to_drop]
-        for table_name in table_names:
-            assert table_name not in self.schema._schema_tables, (
-                f"You are dropping table {table_name} in {self.schema.name} but it is still present"
-                " in the schema"
-            )
-        with self.pipeline._sql_job_client(self.schema) as client:
-            client.drop_tables(*table_names, replace_schema=True)
-            # also delete staging but ignore if staging does not exist
-            if isinstance(client, WithStagingDataset):
-                with contextlib.suppress(DatabaseUndefinedRelation):
-                    with client.with_staging_dataset():
-                        client.drop_tables(*table_names, replace_schema=True)
```

Comment:
@rudolfix 
We were deleting all schema versions with `replace_schema` flag before, and I think we need to do that, or at least delete all schemas that contain the tables being dropped, not just the "current version".   
I might be wrong there but think otherwise we can on later run accidentally re-create a schema that's identical to some previous version it will think those table already exist.   

I thought it was kinda drastic to erase history completely like that, but maybe it's a better workaround? Should we work on properly supporting "subtractive" schema migrations? It is a purely additive versioning system basically so it's all a bit hacky imo.

Reply:
OK you are right. still I prefer to drop all schemas (with given name!). we just need to document this and maybe use :::caution::: for that